clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -346,4 +347,52 @@ public AlterConfigsResult alterConfigs(Map<ConfigResource, Config> configs) {
      * @return                The AlterConfigsResult
      */
     public abstract AlterConfigsResult alterConfigs(Map<ConfigResource, Config> configs, AlterConfigsOptions options);
+
+    /**
+     * Query the log directory information for the specified log directories.
+     * All log directories on a broker are queried if an empty collection of log directories is specified for this broker
+     *
+     * This operation is supported by brokers with version 0.11.1.0 or higher.
+     *
+     * @param logDirsByBroker     A list of log dirs per broker
+     * @return                    The DescribeDirsResult
+     */
+    public DescribeDirsResult describeDirs(Map<Integer, Collection<String>> logDirsByBroker) {
+        return describeDirs(logDirsByBroker, new DescribeDirsOptions());
+    }
+
+    /**
+     * Query the log directory information for the specified log directories.
+     * All log directories on a broker are queried if an empty collection of log directories is specified for this broker
+     *
+     * This operation is supported by brokers with version 0.11.1.0 or higher.
+     *
+     * @param logDirsByBroker     A list of log dirs per broker
+     * @param options             The options to use when querying log dir info
+     * @return                    The DescribeDirsResult
+     */
+    public abstract DescribeDirsResult describeDirs(Map<Integer, Collection<String>> logDirsByBroker, DescribeDirsOptions options);
```

Comment:
I think `LogDir` may be a bit more verbose than Dir. It seems OK to be use `dir` instead of `logDir` as long as logDir will be the only "dir". If it is necessary to use `dir` instead of `logDir`, do you think we should rename `DescribeDirsRequest` to `DescribeLogDirsRequest`?